fe/fe-core/src/main/java/org/apache/doris/statistics/HistogramTask.java

```diff
@@ -48,7 +49,16 @@ public class HistogramTask extends BaseAnalysisTask {
             + "    HISTOGRAM(`${colName}`, 1, ${maxBucketNum}) AS buckets, "
             + "    NOW() AS create_time "
             + "FROM "
-            + "    `${dbName}`.`${tblName}` TABLESAMPLE (${percentValue} PERCENT)";
+            + "    `${dbName}`.`${tblName}`";
+
+    /** To avoid too much data, use the following efficient sampling method */
+    private static final String ANALYZE_HISTOGRAM_SQL_TEMPLATE_PART = ANALYZE_HISTOGRAM_SQL_TEMPLATE
+            + "    PARTITION (${partName})"
+            + "    TABLESAMPLE (${percentValue} PERCENT)";
+
+    /** To avoid too much data, use the following efficient sampling method */
+    private static final String ANALYZE_HISTOGRAM_SQL_TEMPLATE_FULL = ANALYZE_HISTOGRAM_SQL_TEMPLATE
```

Comment:
Why full collection also based on table sample function

Reply:
yeah, this should be in the sample task(Next pr)